packages/angular/cli/lib/init.ts

```diff
@@ -79,14 +79,14 @@ let forceExit = false;
     }
 
     const rawCommandName = process.argv[2];
-    // When using the completion command, don't show the warning as otherwise this will break completion.
-    if (isGlobalGreater && rawCommandName !== 'completion') {
-      // If using the update command and the global version is greater, use the newer update command
-      // This allows improvements in update to be used in older versions that do not have bootstrapping
+    if (isGlobalGreater) {
+      // If using the update/completion command and the global version is greater. Use the global version.
+      // - update: This allows improvements in update to be used in older versions that do not have bootstrapping.
+      // - completion: This allows improvements to the completion command and handles the case where older local
+      //   versions of the CLI don't have this command.
       if (
-        rawCommandName === 'update' &&
-        cli.VERSION &&
-        cli.VERSION.major - globalVersion.major <= 1
+        rawCommandName === 'completion' ||
```

Comment:
I think we should consider limiting this to versions that don't have the command (< 14) so that differences in future versions don't lead to discrepancies in completion behavior.